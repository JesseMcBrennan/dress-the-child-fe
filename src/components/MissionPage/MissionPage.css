.partner-logos {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  width: auto;
  height: 5rem;
  margin: 0px 25rem; }

.partner-logo {
  width: 5rem;
  height: 5rem; }

.rotary-logo {
  background: no-repeat center/4rem url("../../images/rotary-logo.png"); }

.school-logo {
  background: no-repeat center/4rem url("../../images/school-logo.png"); }
