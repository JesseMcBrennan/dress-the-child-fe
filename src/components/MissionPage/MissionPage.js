import React from 'react';
import './MissionPage.css';

export const MissionPage = () => (
  <div className="mission-card">
    <h1 className="main-title">Our Mission</h1>
    <hr className="horizontal-rule" />
    <p className="main-body">
      The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The
      Child program is to provide new clothing and shoes for children living in
      the poorest families in Dona Ana County. Many of these children have never
      had new clothes and have never had the opportunity to shop for clothes
      that might be appealing to them. The founders and program volunteers
      believe that children whom do not have adequate clothing can’t help but
      have lower self esteem than other children and tend to perform poorly in
      school and suffer from other social issues. We believe that by providing
      these children with new clothes, these kids will have better self worth,
      an improved opportunity for scholastic success, and have an opportunity
      for a better life.
    </p>
    <div className="partner-logos">
      <div className="partner-logo rotary-logo" />
      <div className="partner-logo school-logo" />
    </div>
  </div>
);

export default MissionPage;
