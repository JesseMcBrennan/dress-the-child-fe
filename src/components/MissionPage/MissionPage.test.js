import React from 'react';
import { shallow } from 'enzyme';
import { MissionPage } from './MissionPage';

describe('MissionPage', () => {
  let mockLocation;
  let wrapper;
  beforeEach(() => {
    mockLocation = {
      pathname: '/mission'
    };
    wrapper = shallow(<MissionPage location={mockLocation} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
