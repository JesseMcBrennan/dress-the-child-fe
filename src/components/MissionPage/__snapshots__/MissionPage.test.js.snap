// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MissionPage
    location={
      Object {
        "pathname": "/mission",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title"
        >
          Our Mission
        </h1>,
        <hr
          className="horizontal-rule"
        />,
        <p
          className="main-body"
        >
          The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.
        </p>,
        <div
          className="partner-logos"
        >
          <div
            className="partner-logo rotary-logo"
          />
          <div
            className="partner-logo school-logo"
          />
        </div>,
      ],
      "className": "mission-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Our Mission",
          "className": "main-title",
        },
        "ref": null,
        "rendered": "Our Mission",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "horizontal-rule",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.",
          "className": "main-body",
        },
        "ref": null,
        "rendered": "The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="partner-logo rotary-logo"
            />,
            <div
              className="partner-logo school-logo"
            />,
          ],
          "className": "partner-logos",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "partner-logo rotary-logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "partner-logo school-logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title"
          >
            Our Mission
          </h1>,
          <hr
            className="horizontal-rule"
          />,
          <p
            className="main-body"
          >
            The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.
          </p>,
          <div
            className="partner-logos"
          >
            <div
              className="partner-logo rotary-logo"
            />
            <div
              className="partner-logo school-logo"
            />
          </div>,
        ],
        "className": "mission-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Our Mission",
            "className": "main-title",
          },
          "ref": null,
          "rendered": "Our Mission",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "horizontal-rule",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.",
            "className": "main-body",
          },
          "ref": null,
          "rendered": "The mission of the Las Cruces Rio Grande Rotary Foundation’s Dress The Child program is to provide new clothing and shoes for children living in the poorest families in Dona Ana County. Many of these children have never had new clothes and have never had the opportunity to shop for clothes that might be appealing to them. The founders and program volunteers believe that children whom do not have adequate clothing can’t help but have lower self esteem than other children and tend to perform poorly in school and suffer from other social issues. We believe that by providing these children with new clothes, these kids will have better self worth, an improved opportunity for scholastic success, and have an opportunity for a better life.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="partner-logo rotary-logo"
              />,
              <div
                className="partner-logo school-logo"
              />,
            ],
            "className": "partner-logos",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "partner-logo rotary-logo",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "partner-logo school-logo",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
